### PR TITLE
Add support for compiler flags

### PR DESCRIPTION
**Pull request by pepibumur**
### Short description 📝
I was helping a friend to migrate their app to Tuist and I realized that they have source files with compiler flags, which Tuist doesn't support.

### Solution 📦
Add support for compiler flags to Tuist.

### Implementation 👩‍💻👨‍💻
- [x] Update models accordingly.
- [x] Change the generation logic to set the compiler flags.
- [x] Add and fix tests.

cc @RomainBoulay

### CODE REVIEW

File: Sources/TuistGenerator/Models/Target.swift [hunk 1]
```diff
@@ -3,6 +3,8 @@ import Foundation
 import TuistCore
 
 public class Target: Equatable {
+    public typealias SourceFile = (path: AbsolutePath, compilerFlags: String?)
```
Comment (ollieatkinson) [inline Sources/TuistGenerator/Models/Target.swift]: Do you think this warrants its own type?

File: docs/usage/2-manifest.mdx [hunk 2]
```diff
@@ -212,6 +213,28 @@ let target = Target(name: "App",
...
+default: ""
+}]}/>
+
+<Message info title="ExpressibleByStringLiteral and ExpressibleByArrayLiteral" description="The list of source files can be initialized with a string that represents the glob pattern, or an array of strings, which represents a list of glob patterns. In both cases the comiler flags will have no value."/>
+
+### Source file glob
+
+It represents a glob pattern k
+
+// TODO
```
Comment (ollieatkinson) [inline docs/usage/2-manifest.mdx]: TODO?

**Comment by RomainBoulay** [comment]
Gracias tio @pepibumur <3

File: Sources/TuistGenerator/Models/Target.swift (hunk 1 above)
Comment (pedropinerashopify) [reply Sources/TuistGenerator/Models/Target.swift]: It shortens the declaration of the functions a bit.

**Pull request merged**